pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/MailboxAssignmentVisitor.java

```diff
@@ -59,9 +59,38 @@ public Void process(PlanNode node, DispatchablePlanContext context) {
             receiverMailboxesMap.computeIfAbsent(workerId, k -> new HashMap<>()).put(senderFragmentId, mailboxMetadata);
           }
         });
+      } else if (senderMetadata.isPartitionedTableScan()) {
+        // For partitioned table scan, send the data to the worker with the same worker id (not necessary the same
+        // instance)
```

Comment:
maybe we can add a TODO - which instead of same worker-id we will do a 1:2 or 1:4 fan-out. 
(possibly based on the parallelism setting such as `set stageParallelism=xxx`)

Reply:
Added